File: controledeestoque/src/main/java/com/inventario/imobilizado/model/Emprestante.java
package com.inventario.imobilizado.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.Objects;

@Entity
@Table(name = "emprestante")
@Getter
@Setter
@NoArgsConstructor
@EqualsAndHashCode(of = "id_emprestante")
public class Emprestante {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_emprestante")
    private int id_emprestante;
    @Column(name = "num_identificacao")
    private String num_identificacao;
    @Column(name = "nome")
    private String nome;
    @Column(name = "status_emprestante")
    private String status_emprestante;

    public Emprestante(String num_identificacao, String nome, String status_emprestante) {
        this.num_identificacao = num_identificacao;
        this.nome = nome;
        this.status_emprestante = status_emprestante;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Emprestante that = (Emprestante) o;
        return id_emprestante == that.id_emprestante && num_identificacao == that.num_identificacao && Objects.equals(nome, that.nome) && Objects.equals(status_emprestante, that.status_emprestante);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id_emprestante, num_identificacao, nome, status_emprestante);
    }
}

File: projeto_integrador/lib/models/emprestante.dart
class Emprestante {
  final int id_emprestante;
  final String num_identificacao;
  final String nome;
  final String status_emprestante;

  Emprestante({
    required this.id_emprestante,
    required this.num_identificacao,
    required this.nome,
    required this.status_emprestante,
  });

  factory Emprestante.fromJson(Map<String, dynamic> json) {
    Emprestante emprestante = Emprestante(
      id_emprestante: json['id_emprestante'],
      num_identificacao: json['num_identificacao'],
      nome: json['nome'],
      status_emprestante: json['status_emprestante'],
    );
    return emprestante;
  }

  Map<String, dynamic> toJson() {
    return {
      'id_emprestante': id_emprestante,
      'num_identificacao': num_identificacao,
      'nome': nome,
      'status_emprestante': status_emprestante,
    };
  }
}
